lxd/storage/drivers/utils_cgo.go

```diff
@@ -265,6 +276,7 @@ import "C"
 // LoFlagsAutoclear determines whether the loop device will autodestruct on last
 // close.
 const LoFlagsAutoclear int = C.LO_FLAGS_AUTOCLEAR
+const LoFlagsDirectIO int = C.LO_FLAGS_DIRECT_IO
```

Comment:
So one catch with `LO_FLAGS_DIRECT_IO` is that it's only supported on Linux 4.10 and above.

As a result, I think we'd need `prepare_loop_dev` to check if `LO_FLAGS_DIRECT_IO` is part of its `flags` and if it is then handle the `ioctl` failure which may come if the kernel doesn't support that flag, then strip the flag and retry without it.

@brauner do you know what the `ioctl` would return if unsupported? ENOSYS? EINVAL?

Reply:
I added a separate `ioctl` call using the LOOP_SET_DIRECT_IO operation in the updated commits. The flag is stripped on failure.